### PR TITLE
[4.0] Modify modal window and remove orphan label in privacy consent and terms and conditions fields

### PR DESCRIPTION
**Pull request by drmenzelit**
Pull Request for Issue #34529, #33664 and https://github.com/joomla/joomla-cms/issues/33664#issuecomment-859354781.

### Summary of Changes
Modified the attributes `bodyHeight` and `modalWidth` in `renderModal` in the label layouts of privacy consent and terms and conditions. The values has to be int and not strings.

Furthermore removed the `label` element and related popover. The label was marked as orphaned element by WAVE as stated in the comment linked above.

Removed the description of the fields, it can't be hidden by configuration.

### Testing Instructions

- Activate the plugins System - Privacy Consent and User - Terms and Conditions
- Configure both plugins
- Allow user registration
- Create a menu item for registration form
- Go to the registration form and click on the links to the Privacy Policy and Terms and Conditions, a modal window will open
- Inspect the code, if possible test with WAVE

### Actual result BEFORE applying this Pull Request
See issue #34529 
The modal window is very small

Inspect the code and you will see a label element surrounding the link
See comment https://github.com/joomla/joomla-cms/issues/33664#issuecomment-859354781

### Expected result AFTER applying this Pull Request
No label element, only link if the plugin is correctly configured, only a span if the link is missing

The modal window is bigger now.

### CODE REVIEW

**Comment by ghost** [comment]
### Modal 

Privacy:

image

Terms and Conditions:

image

### Label element:

without pr | with pr
-- | --
image | image
image | image

span with pull request:
image

**Comment by drmenzelit** [comment]
Thank you @sandramay0905 for the screenshots :-)

**Comment by Kostelano** [comment]
In a modal window with privacy, you need an indent from the block of information to the form with fields.

**Comment by drmenzelit** [comment]
> 
> 
> In a modal window with privacy, you need an indent from the block of information to the form with fields.

sorry, but I don't understand this, can you explain?

**Comment by Kostelano** [comment]
I have not tested, but paid attention to the screenshots (below is what I am talking about).

Screenshot_1

Sorry, maybe this is outside the scope of this PR.

**Comment by drmenzelit** [comment]
Ah... that is not related to this PR, I think @sandramay0905 took a random menu item to test :-)

File: layouts/plugins/system/privacyconsent/label.php [hunk 1]
```diff
@@ -60,35 +60,15 @@
...
-	HTMLHelper::_('bootstrap.popover', '.hasPopover', ['trigger' => 'hover focus']);
-	$label .= ' title="' . htmlspecialchars(trim($text, ':'), ENT_COMPAT, 'UTF-8') . '"';
-	$label .= ' data-bs-content="' . htmlspecialchars(
-		$translateDescription ? Text::_($description) : $description,
-		ENT_COMPAT,
-		'UTF-8'
-	) . '"';
-}
-
-if (Factory::getLanguage()->isRtl())
```
Comment (Quy) [inline layouts/plugins/system/privacyconsent/label.php:82]: Please remove `use Joomla\CMS\Factory; `

**Comment by brianteeman** [comment]
Does this also address #33664

**Comment by drmenzelit** [comment]
@brianteeman only partially, the semantic is hopefully better now, but I was not able to find a way to hide the field description

**Comment by alikon** [comment]
closed #34529...
in the mean time :smile:

**Comment by brianteeman** [comment]
Question: Shouldnt the modal load without any page navigation

**Comment by brianteeman** [comment]
I think you can just delete the description. It doesnt add anything and its not displayed conditionally on the presence of a link.

**Comment by drmenzelit** [comment]
> 
> 
> Question: Shouldnt the modal load without any page navigation

There is no page navigation in the modal, or am I missing something?

**Comment by drmenzelit** [comment]
> 
> 
> I think you can just delete the description. It doesnt add anything and its not displayed conditionally on the presence of a link.

The description is a language string and loaded in the field (https://github.com/joomla/joomla-cms/blob/4.0-dev/plugins/system/privacyconsent/forms/privacyconsent.xml)

**Comment by brianteeman** [comment]
I know why and where the language string is used. I am saying that it can be deleted from both the ini and the xml.

**Comment by brianteeman** [comment]
Re page navigation I meant this
image

**Comment by drmenzelit** [comment]
oh, I don't know if it possible to hide the navigation when a page is opened in a modal window...

**Comment by drmenzelit** [comment]
> 
> 
> I know why and where the language string is used. I am saying that it can be deleted from both the ini and the xml.

Thank you for confirming we can delete that.

**Comment by Kostelano** [comment]
Do we need a similar message for the Terms of Service? If we DO NOT select material in the User - Profile (TOS parameters) plugin settings, then we just get a phrase without a link and without additional. block of information (like the one below in the screenshot). We must post Yes, it is not clear why and for what.

Screenshot_1

**Comment by drmenzelit** [comment]
@Kostelano I'm afraid the Terms of Service is out of the scope of this PR. The profile plugin needs to be extended with a message field for TOS like in Privacy Consent or Terms & Conditions plugin. I'm not sure if my programming skills are good enough to do that. But that should be a separate PR.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 2232d03b13eda60900a022c73614874460c1904e<br /><br />Ok, thanks. Then everything is fine for me.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34537">issues.joomla.org/tracker/joomla-cms/34537</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 2232d03b13eda60900a022c73614874460c1904e<br /><br />Also tested WAVE<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34537">issues.joomla.org/tracker/joomla-cms/34537</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34537">issues.joomla.org/tracker/joomla-cms/34537</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**